app/daemon/daemon.go

```diff
@@ -100,17 +101,91 @@ type GitHubReleaseData struct {
 }
 
 
-func DownloadAndStart(serverConfig *conf.ServerConfig, userConfig *conf.Config) (*exec.Cmd) {
+var runningDaemon *exec.Cmd
+var minHeartbeat int64 = 500 // the lowest value the hb checker can be set to
+
+
+// StartManager is a simple system that checks if
+// the daemon is alive. If not it tries to start it
+func StartManager()  {
+
+	// set the heartbeat interval but make sure it is not
+	// less than the min heartbeat setting
+	hbInterval := minHeartbeat
+	if  conf.ServerConf.DaemonHeartbeat > hbInterval {
+		hbInterval = conf.ServerConf.DaemonHeartbeat
+	}
+
+
+	ticker := time.NewTicker(time.Duration(hbInterval) * time.Millisecond)
+	go func() {
+		for t := range ticker.C {
+
+			log.Println(t)
+
+			// check to see if the daemon is alive
+			if isAlive() {
+				//log.Println("daemon is alive....")
+			} else {
+
+				log.Println("Deamon is unresponsive")
+
+				if runningDaemon != nil {
+					Stop(runningDaemon)
+				}
+
+
+				//Start the daemon and download it if necessary
+				cmd, err := DownloadAndStart(conf.ServerConf, conf.UserConf)
+
+				if err != nil {
+					log.Println(err)
+				} else {
+					runningDaemon = cmd
+				}
+
+			}
+
+		}
+	}()
+}
+
+
+// isAlive performs a simple rpc command to the daemon
+// returns false on error
+func isAlive () bool {
+
+	isLiving := true
+
+	n := deamonrpc.RpcRequestData{}
+	n.Method = "getblockcount"
+
+	_, err := deamonrpc.RequestDaemon(n, conf.UserConf)
+
+	if err != nil {
+		isLiving = false
+	}
+
+	return isLiving
+
+}
+
+
+func DownloadAndStart(serverConfig conf.ServerConfig, userConfig conf.UserConfig) (*exec.Cmd, error) {
```

Comment:
whats with the double brackets? () ()

Reply:
first brackets is the function params that will be passed in, the second is the return types that will be passed back